golem_core/demand_builder/model.py

```diff
@@ -0,0 +1,304 @@
+import abc
+import inspect
+import enum
+import dataclasses
+import datetime
+
+from typing import (
+    Any,
+    Dict,
+    List,
+    Type,
+    TypeVar,
+    Final,
+    Literal,
+    Tuple,
+    Iterable,
+)
+
+TModel = TypeVar("TModel", bound="Model")
+Props = Dict[str, str]
+
+PROP_KEY: Final[str] = "key"
+PROP_OPERATOR: Final[str] = "operator"
+PROP_MODEL_FIELD_TYPE: Final[str] = "model_field_type"
+
+ConstraintOperator = Literal["=", ">=", "<="]
+ConstraintGroupOperator = Literal["&", "|", "!"]
+
+
+class ConstraintException(Exception):
+    pass
+
+
+class InvalidPropertiesError(Exception):
+    """Raised by `Model.from_properties(cls, properties)` when given invalid `properties`."""
+
+
+class ModelFieldType(enum.Enum):
+    constraint = "constraint"
+    property = "property"
+
+
+@dataclasses.dataclass
+class Model(abc.ABC):
```

Comment:
Name that was used in yapapi... I have no better idea to pick.
Its not `Demand` related, it can contain properties and constraints... but its only golem-related.
I don't know if `GolemBaseModel` would be a better option.

Reply:
`ComputingResourceModel`